### ISSUE TITLE
Crowdfund App Perks with custom price are not taken into account of goal (LNURL method at least)

### ISSUE THREAD

**Issue by Perlover** [btcpayserver/btcpayserver#3746]
**Describe the bug**

Crowdfund App Perks with custom price are not taken into account of goal (LNURL method at least)

**To Reproduce the bug**
Steps to reproduce the reported bug:
1. Create Crowdfund App
2. Create some Perks with sums
3. Create one Perk with "Price Custom"
4. To go Crowdfund page
5. Click by Custom Perk
6. To pay a Custom Perk by Lightning for example (I didn't test Bitcoin on-chain) by LNURL method
7. When a Custom Perk is paid the sum is not taken into account - a goal volume is not increased and the amount is visible only in the general invoices, but is not visible in the invoices of Crowdfunding itself!

**Expected behavior**

The custom price contribution should be taken into account both in the amount of the goal and visible in invoices of crowdfunding admin area, for example.

**Your BTCPay Environment (please complete the following information):**
- BTCPay Server Version:    1.5.2.0
- Deployment Method:        LunaNode's template
- Browser:                  Firefox 100.*

**Logs (if applicable)**

**Setup Parameters**

**Additional context**

**Comment by pavlenex** [comment]
LNURL by design doesn't include the amount, after all you did define "custom amount" in the perk. I couldn't replicate the other issue you're reporting, when I pay the LNURL, It is reflected in my crowdfunding.

@Kukks I think we really should ditch lnurl-pay from the checkout page and use the bolt11 zero invoices for zero amount invoices. It just doesn't make sense to use it the way it is set up now.

**Comment by Kukks** [comment]
> LNURL by design doesn't include the amount, after all you did define "custom amount" in the perk. I couldn't replicate the other issue you're reporting, when I pay the LNURL, It is reflected in my crowdfunding.
> 
> @Kukks I think we really should ditch lnurl-pay from the checkout page and use the bolt11 zero invoices for zero amount invoices. It just doesn't make sense to use it the way it is set up now.

I disagree as it disables other features. This is out fo scope for this issue. We can discuss it next week.

**Comment by pavlenex** [comment]
What other features, we discussed this already, even fiatjeff said it does not make sense in the invoice checkout. Indeed out of scope, but evidently it causes confusion for people.

**Comment by Kukks** [comment]
> What other features? We discussed this already, even fiatjeff said it does not make sense in the invoice checkout. Indeed out of scope, but evidently it causes confusion for people.

I have not seen that remark by him, curious to see that reasoning. 0 amount invoice support is not very good, only very few wallets support it and I think our handling of different payment amounts to a bolt11  may not even register it in case of over payment.

**Comment by Perlover** [comment]
After all, the problem that I noticed many times was somehow lost behind the topic of zero invoices. I want to say right away - LNURL-pay does not cause me confusion. The problem is different. When I pay Perk with "Price Custom" - it is not added to the amount collected by the Goal. That is, I pay the invoice, but the amount is not counted.

[You can check yourself...](https://pay.fuckedbitcoin.com/apps/3NSbM4hYAqqP1tvZMJFWMJjBqG5t/crowdfund)

**Comment by pavlenex** [comment]
@Perlover Thank you for filing a bug, I was able to reporduce what you're explaining.

@Kukks when user generates the invoice by selecting `lnurl` from the tab, crowdfund just doesn't account for it. Two solutions:
- Remove this way of generating invoice, I don't think that's it's beneficial, let's just generate an invoice
- Fix it and leave it as it is


To replicate the issue:
1. Create crowdfunding
2. Add custom amount perk
3. Click on contribute and choose `lnurl` 
![Screenshot 2022-05-28 at 19 59 54](https://user-images.githubusercontent.com/36959754/170837578-ffc99aef-b866-41e1-a779-f27b96c5b667.png)
4. Pay the invoice, nothing happens in crowdfunding.

**Comment by pavlenex** [comment]
Personally, I would prefer if we move out the ability for user to pick between standard and lnurl and just leave it to happen on the invoice level, this is way too many clicks for to achieve the same thing.

**Comment by Kukks** [comment]
> Personally, I would prefer if we move out the ability for user to pick between standard and lnurl and just leave it to happen on the invoice level, this is way too many clicks for to achieve the same thing.

Considering you also want to remove lnurl from the invoice, you will end up removing all lnurl functionality at this rate.

**Comment by Kukks** [comment]
It seems like this is probably an internal bug around how crowdfund on detects the goal progress

**Comment by pavlenex** [comment]
> > Personally, I would prefer if we move out the ability for user to pick between standard and lnurl and just leave it to happen on the invoice level, this is way too many clicks for to achieve the same thing.
> 
> Considering you also want to remove lnurl from the invoice, you will end up removing all lnurl functionality at this rate. 

Removing LNURL-pay from the invoice is something LNURL creator suggested. I was just questioning if we need it. The LNURL in the crowdfund to generate the invoice feels like unnecessary step since it can already be done on the invoice level.

**Comment by Perlover** [comment]
> Maybe it is the different bindings (crowdfund-app_ vs pos-app_) to the app that play a role in this?

And now I see in BTCPay admin area (Payment -&gt; Invoices) - I see the Order ID column there. There are listed "pos-app_3NSbM4hYAqqP1tvZMJFW..." **not linked** strings and the **linked** to crowdfinding company "crowdfund-app_3NSbM4hYAqqP1tvZMJFW.." strings. By all indications, the BTCPay internally does not associate a "pos-app" prefix with a fundraising company.

**Comment by bolatovumar** [comment]
Made a PR to just remove this way of generating the LNURL QR code since it doesn't really make sense to have two ways to do the same thing and pretty much everywhere else we just use the invoice checkout page for this purpose.

**Comment by Kukks** [comment]
Yes, it's exactly the same view so using the sidebar vs modal is identical in functionality. 
Just tried again: https://i.imgur.com/wd87nAU.gif

**Comment by pavlenex** [comment]
Closing  due to #3912
